src/main/java/com/cinema/dao/impl/UserDaoImpl.java

```diff
@@ -19,7 +19,8 @@ public UserDaoImpl(SessionFactory sessionFactory) {
     @Override
     public Optional<User> findByEmail(String email) {
         try (Session session = sessionFactory.openSession()) {
-            return session.createQuery("FROM User WHERE email=:email", User.class)
+            return session.createQuery("FROM User u "
```

Comment:
use inner join here, there should not be a situation when you have user without roles

Reply:
done